### PR TITLE
Use @bazel_tools//tools/bash/runfiles for runfiles resolve

### PR DESCRIPTION
**Pull request by gregmagolan**
This function can used by any bash script that needs to resolve runfiles including many scripts in rules_typescript.

### CODE REVIEW

**Comment by alexeagle** [comment]
Should we be using //src/tools/runfiles:runfiles_sh_lib instead which @laszlocsomor added?

**Comment by alexeagle** [comment]
@gregmagolan - see https://github.com/bazelbuild/bazel/issues/4460#issuecomment-388868234

**Comment by alexeagle** [comment]
@gregmagolan Bazel 0.14 was released today so this is unblocked

File: internal/node/node_launcher.sh [hunk 1]
```diff
@@ -15,6 +15,30 @@
 
 set -e
 
+# --- begin runfiles.bash initialization ---
```
Comment (alexeagle) [inline internal/node/node_launcher.sh]: Seems like not less code than before... 😥

**Comment by gregmagolan** [comment]
PR updates to ngcontainer with bazel 0.14.0 and includes skylint fixes but also disables deprecated-api skylint checks since the legacy `typescript` provider can't be updated to a modern provider without breaking compatability

File: internal/node/node_launcher.sh (hunk 1 above)
Comment (gregmagolan) [reply internal/node/node_launcher.sh]: Sadly no. But at least its the recommended approach.

File: WORKSPACE [hunk 2]
```diff
@@ -26,6 +86,8 @@ local_repository(
     path = "examples/packages",
 )
 
+load("//:defs.bzl", "node_repositories")
```
[the remark this answers is not in the document]
Comment (gregmagolan) [reply WORKSPACE]: used in CI
```
      - run: bazel test --config=ci @program_example//...
      - run: bazel test --config=ci @packages_example//...
```

File: internal/node/node_repositories.bzl [hunk 3]
```diff
@@ -423,8 +423,8 @@ def node_repositories(
       files need to be included in your node_modules filegroup. This option is desirable as it gives a stronger
       guarantee of hermiticity which is required for remote execution.
   """
-  # Windows users need sh_binary wrapped as an .exe
-  check_bazel_version("0.5.4")
+  # @bazel_tools//tools/bash/runfiles is required for nodejs
+  check_bazel_version("0.14.0")
```
[the remark this answers is not in the document]
Comment (gregmagolan) [reply internal/node/node_repositories.bzl]: right. good catch thanks

**Pull request merged**